ci/install_protobuf.sh

```diff
@@ -31,12 +31,41 @@ set -e
 # when calling this script
 #
 
-export CPP_PROTOBUF_VERSION="3.${PROTOBUF_VERSION}"
-
-cd /
-wget https://github.com/google/protobuf/releases/download/v${PROTOBUF_VERSION}/protobuf-cpp-${CPP_PROTOBUF_VERSION}.tar.gz
-tar zxf protobuf-cpp-${CPP_PROTOBUF_VERSION}.tar.gz --no-same-owner
-cd protobuf-${CPP_PROTOBUF_VERSION}
-./configure
-make -j $(nproc) && make install
+CPP_PROTOBUF_BUILD_OPTIONS=(
+  "-DCMAKE_POSITION_INDEPENDENT_CODE=ON"
+  "-Dprotobuf_BUILD_TESTS=OFF"
+  "-Dprotobuf_BUILD_EXAMPLES=OFF"
+  "-Dprotobuf_MODULE_COMPATIBLE=ON"
+)
+
+if [ ! -z "${CXX_STANDARD}" ]; then
+    CPP_PROTOBUF_BUILD_OPTIONS=(${CPP_PROTOBUF_BUILD_OPTIONS[@]} "-DCMAKE_CXX_STANDARD=${CXX_STANDARD}")
+fi
+
+if [[ ${PROTOBUF_VERSION/.*/} -ge 22 ]]; then
+  CPP_PROTOBUF_BUILD_OPTIONS=(${CPP_PROTOBUF_BUILD_OPTIONS[@]} "-Dprotobuf_MODULE_COMPATIBLE=ON")
+fi
+
+if [[ ${PROTOBUF_VERSION/.*/} -ge 22 ]]; then
+  export CPP_PROTOBUF_VERSION="${PROTOBUF_VERSION}"
+  CPP_PROTOBUF_PACKAGE_NAME="protobuf-${CPP_PROTOBUF_VERSION}"
```

Comment:
Please add a comment to explain _why_ we need this.

Something like:
- protobuf 22.xxx ships packages named ???
- protobuf 3.yyy ships packages named ???

Also, please clarify if we should pick:

https://github.com/protocolbuffers/protobuf/releases/download/v22.3/protobuf-22.3.tar.gz

or

https://github.com/protocolbuffers/protobuf/archive/refs/tags/v22.3.tar.gz

If there is a difference, detail why.

If this is the same, can the following URL pattern work for both 22.xxx and 3.yyy ?

https://github.com/protocolbuffers/protobuf/archive/refs/tags/v22.3.tar.gz

I admit I am totally confused here by the protobuf version names, hence comments explaining what probuf is doing will help.

Reply:
It seems 3.XX.YY/4.XX.YY is alias for XX.YY. And there will be no 3.XX.YY after 22.
And the released source packages are moved into the tag of XX.YY and not 4.XX.YY after 22.
Also, there is only protobuf-22.3.tar.gz, and there is no protobuf-cpp-22.3.tar.gz any more.